src/mockup.py

```diff
@@ -182,40 +193,92 @@ def long_term_total_payoff(opening_payoffs, exp_p, delta):
     return payoffs + exp_p * delta ** len(opening_payoffs) / (1.0-delta)
 
 # %%
+def make_history(player1_moves, player2_mem1_strategy):
+    """
+    play the game between two players for `len(player1_moves)` turns and return the history
+    player1_moves: list of moves of player1
+    player2_mem1_strategy: (init, pcc, pcd, pdc, pdd)  (tuple of Actions)
+    """
+    last = (player1_moves[0], player2_mem1_strategy[0])
+    history = []
+    history.append(last)
+    for i in range(1,len(player1_moves)):
+        m1 = player1_moves[i]
+        if last == (C,C):
+            m2 = player2_mem1_strategy[1]
+        elif last == (C,D):
+            # from player2's viewpoint, this is (D,C)
+            m2 = player2_mem1_strategy[3]
+        elif last == (D,C):
+            # from player2's viewpoint, this is (C,D)
+            m2 = player2_mem1_strategy[2]
+        elif last == (D,D):
+            m2 = player2_mem1_strategy[4]
+        else:
+            raise ValueError("Invalid history")
+        history.append((m1, m2))
+        last = (m1, m2)
+    return history
+
+def calc_payoff_from_history(history, benefit, cost):
```

Comment:
Should we make this function more generic? Should it take R, S, T, P as arguments?